### PR TITLE
Limit max eager activities to 3

### PR DESCRIPTION
**Pull request by bergundy**
See https://github.com/temporalio/sdk-java/pull/1455

### CODE REVIEW

File: core/src/worker/workflow/mod.rs [hunk 1]
```diff
@@ -475,6 +477,11 @@ impl Workflows {
                         if same_task_queue {
                             if let Some(p) = at_handle.reserve_slot() {
                                 reserved.push(p);
+                                if reserved.len()
+                                    >= MAX_EAGER_ACTIVITY_RESERVATIONS_PER_WORKFLOW_TASK
+                                {
+                                    break;
+                                }
```
Comment (Sushisource) [inline core/src/worker/workflow/mod.rs:484]: This won't work. Commands come in with `request_eager_execution` set true. You need to set it false for the remaining commands in the list.

A test should prove this works

Comment (bergundy) [reply core/src/worker/workflow/mod.rs:484]: Oh, I forgot about that part, thanks

**Comment by bergundy** [comment]
Fixed @Sushisource

File: core/src/core_tests/activity_tasks.rs [hunk 2]
```diff
@@ -658,15 +662,27 @@ async fn activity_tasks_from_completion_are_delivered() {
...
+                        task_token: vec![2],
+                        activity_id: "act_id_2_same_queue".to_string(),
+                        ..Default::default()
+                    },
+                    PollActivityTaskQueueResponse {
+                        task_token: vec![3],
+                        activity_id: "act_id_3_same_queue".to_string(),
+                        ..Default::default()
+                    },
+                ],
```
Comment (Sushisource) [inline core/src/core_tests/activity_tasks.rs:681]:
```suggestion
                activity_tasks: (1..3).map(|i| 
                    PollActivityTaskQueueResponse {
                        task_token: vec![i],
                        activity_id: format!("act_id_{i}_same_queue"),
                        ..Default::default()
                    }).collect()
```

**Review by Sushisource — APPROVED** [review]
Would be nice to save the copy-pastes

**Comment by bergundy** [comment]
I hope I made you happy with the last commit @Sushisource

**Pull request merged**